### PR TITLE
Feature/dcc 5572 sodalite start client

### PR DESCRIPTION
**Pull request by KevinFHartmann**

### CODE REVIEW

File: sodalite-client/src/main/java/org/icgc/dcc/sodalite/client/command/Command.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+   * Otherwise, returns a Command object capable of running the given subcommand.
+   */
+  public static Command parse(String[] args) {
+
+    if (args.length == 0) {
+      return new ErrorCommand("No subcommand given");
+    }
+    String cmd = args[0];
+    // log.info("Looking up command " + cmd);
+    Command c = commands.getOrDefault(cmd, new ErrorCommand("Unknown subcommand: " + cmd));
```
Comment (andricDu) [inline sodalite-client/src/main/java/org/icgc/dcc/sodalite/client/command/Command.java]: val

Unless you plan or replacing all this infrastructure later with JCommander.

File: sodalite-client/src/main/java/org/icgc/dcc/sodalite/client/register/Registry.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestTemplate;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import lombok.SneakyThrows;
+import lombok.val;
+
+/**
+ *
```
Comment (andricDu) [inline sodalite-client/src/main/java/org/icgc/dcc/sodalite/client/register/Registry.java]: Add comment or delete this

File: sodalite-client/src/main/resources/application.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+# TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+# OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+# IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+# DCC Client Server - Default Configuration
+#
+
+server:
+  port: 9090
```
Comment (andricDu) [inline sodalite-client/src/main/resources/application.yml]: clean up the config here. A lot of the server configuration is present for the client config. Importantly it should look similar to your config classes.

File: sodalite-client/src/main/java/org/icgc/dcc/sodalite/client/command/RegisterCommand.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+    val config = new SodaliteConfig();
+    registry = new Registry(config);
+    String uploadId = args[1];
+
+    String result = registry.registerAnalysis(uploadId, json);
+    output(result);
+
+    return this.status;
+
```
Comment (andricDu) [inline sodalite-client/src/main/java/org/icgc/dcc/sodalite/client/command/RegisterCommand.java]: delete this newline

File: sodalite-client/src/main/java/org/icgc/dcc/sodalite/client/command/Status.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  }
+
+  public void reportOutput() {
+    System.out.println(outputs);
+  }
+
+  public void report() {
+    reportOutput();
+    reportErrors();
+  }
```
Comment (andricDu) [inline sodalite-client/src/main/java/org/icgc/dcc/sodalite/client/command/Status.java]: add newline after this brace

File: sodalite-client/src/main/java/org/icgc/dcc/sodalite/client/model/ManifestEntry.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+package org.icgc.dcc.sodalite.client.model;
+
+/**
+ * 
+ */
+public class ManifestEntry {
+
+  String fileId;
+  String fileName;
+  String md5sum;
```
Comment (andricDu) [inline sodalite-client/src/main/java/org/icgc/dcc/sodalite/client/model/ManifestEntry.java]: add newline between last statement and last brace

File: sodalite-client/src/main/java/org/icgc/dcc/sodalite/client/command/Command.java (hunk 1 above)
Comment (KevinFHartmann) [inline sodalite-client/src/main/java/org/icgc/dcc/sodalite/client/command/Command.java]: I'll have to check out JCommander first, so I'll make the suggested changes. :-)

**Pull request merged**